### PR TITLE
Add runtime filter docs

### PR DESCRIPTION
**Pull request by ran-huang**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v7.3 (TiDB 7.3 versions)
- [ ] v7.2 (TiDB 7.2 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/14517
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: runtime-filter.md [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+### Step 1: Create a table with TiFlash replicas
+
+Add a TiFlash replica to each of the `catalog_sales` table and the `date_dim` table.
+
+```sql
+ALTER TABLE catalog_sales SET tiflash REPLICA 1;
+ALTER TABLE date_dim SET tiflash REPLICA 1;
+```
+
+等待一段时间，并检查两个表的 TiFlash 副本已准备就绪，即副本的 `AVAILABLE` 字段和 `PROGRESS` 字段均为 `1`。
```
Comment (ran-huang) [inline runtime-filter.md:93]:
```suggestion
```

File: runtime-filter.md [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
++---------------+                   +----------------+
+```
+
+*(The above figure omits the exchange node and other nodes.)*
+
+Runtime Filter 的执行方式如下：
+
+The execution process of Runtime Filter is as follows:
+
+1. Scan the data of `date_dim`, and PhysicalHashJoin calculates a filter condition based on the `date_dim` data, such as `date_dim in (2001/01/01~2001/12/31)`.
```
Comment (qiancai) [inline runtime-filter.md:55]: Add a comment to https://github.com/pingcap/docs-cn/pull/14517#discussion_r1280477767

File: runtime-filter.md [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+    5000    |    |3. Send RF                     |      365
+            |    | filter data                   |
+            |    |                               |
+      +-----+----v------+                +-------+--------+
+      |  TableFullScan  |                | TabelFullScan  |
+      |  store_sales    |                |    date_dim    |
+      +-----------------+                +----------------+
+```
+
+From the above two figures, you can see that the amount of data scanned by `store_sales` is reduced from 1 million to 5000. By reducing the amount of data scanned by Table Full Scan, Runtime Filter can reduce the number of times to match the Hash Table, avoid unnecessary I/O and network transmission, and significantly improve the efficiency of the Join operation.
```
Comment (qiancai) [inline runtime-filter.md:74]:
```suggestion
From the above two figures, you can see that the amount of data scanned by `store_sales` is reduced from 1 million to 5000. By reducing the amount of data scanned by Table Full Scan, Runtime Filter can reduce the number of times to match the hash table, avoid unnecessary I/O and network transmission, and significantly improve the efficiency of the Join operation.
```

File: runtime-filter.md [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+
+## Use Runtime Filter
+
+To use Runtime Filter, you only need to create a table with TiFlash replicas and set [`tidb_runtime_filter_mode`](/system-variables.md#tidb_runtime_filter_mode-introduced-in-v720) to `LOCAL`.
+
+Taking the TPC-DS dataset as an example, this section uses the `catalog_sales` table and the `date_dim` table for Join operations to illustrate how to use Runtime Filter to improve query efficiency.
+
+### Step 1: Create a table with TiFlash replicas
```
Comment (qiancai) [inline runtime-filter.md:82]:
```suggestion
### Step 1. Create TiFlash replicas for tables to be joined
```

File: runtime-filter.md [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+
+mysql> SELECT * FROM INFORMATION_SCHEMA.TIFLASH_REPLICA WHERE TABLE_NAME='date_dim';
++--------------+------------+----------+---------------+-----------------+-----------+----------+
+| TABLE_SCHEMA | TABLE_NAME | TABLE_ID | REPLICA_COUNT | LOCATION_LABELS | AVAILABLE | PROGRESS |
++--------------+------------+----------+---------------+-----------------+-----------+----------+
+| tpcds50      | date_dim   |     1015 |             1 |                 |         1 |        1 |
++--------------+------------+----------+---------------+-----------------+-----------+----------+
+```
+
+### Step 2: Enable Runtime Filter
```
Comment (qiancai) [inline runtime-filter.md:109]:
```suggestion
### Step 2. Enable Runtime Filter
```

File: runtime-filter.md [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
++--------------------------+-------+
+| Variable_name            | Value |
++--------------------------+-------+
+| tidb_runtime_filter_mode | LOCAL |
++--------------------------+-------+
+```
+
+If the value of the system variable is `LOCAL`, Runtime Filter is enabled.
+
+### Step 3: Execute the query
```
Comment (qiancai) [inline runtime-filter.md:130]:
```suggestion
### Step 3. Execute the query
```

File: runtime-filter.md [hunk 7]
```diff
@@ -0,0 +1,257 @@
...
+
+Before executing the query, check the query plan. Use the [`EXPLAIN` statement](/sql-statements/sql-statement-explain.md) to check whether Runtime Filter has taken effect.
```
Comment (qiancai) [inline runtime-filter.md:132]:
```suggestion
Before executing the query, use the [`EXPLAIN` statement](/sql-statements/sql-statement-explain.md) to show the query plan and check whether Runtime Filter has taken effect.
```

File: runtime-filter.md [hunk 8]
```diff
@@ -0,0 +1,257 @@
...
+```
+
+### Step 4: Performance comparison
+
+This example uses the 50 GB TPC-DS data. After Runtime Filter is enabled, the query time is reduced from 0.38 seconds to 0.17 seconds, and the efficiency is improved by 50%. You can use the `ANALYZE` statement to view the execution time of each operator after Runtime Filter takes effect.
+
+The following is the execution information of the query when Runtime Filter is not enabled:
+
+```sql
+mysql> EXPLAIN ANALYZE SELECT cs_ship_date_sk FROM catalog_sales, date_dim WHERE d_date = '2002-2-01' AND cs_ship_date_sk = d_date_sk;
```
Comment (qiancai) [inline runtime-filter.md:181]:
```suggestion
EXPLAIN ANALYZE SELECT cs_ship_date_sk FROM catalog_sales, date_dim WHERE d_date = '2002-2-01' AND cs_ship_date_sk = d_date_sk;
```

File: runtime-filter.md [hunk 9]
```diff
@@ -0,0 +1,257 @@
...
+
+Now, execute the SQL query, and Runtime Filter is applied.
+
+```sql
+SELECT cs_ship_date_sk FROM catalog_sales, date_dim
+WHERE d_date = '2002-2-01' AND
+     cs_ship_date_sk = d_date_sk;
+```
+
+### Step 4: Performance comparison
```
Comment (qiancai) [inline runtime-filter.md:174]:
```suggestion
### Step 4. Performance comparison
```

File: runtime-filter.md [hunk 10]
```diff
@@ -0,0 +1,255 @@
...
+```
+
+By comparing the execution information of the two queries, you can find the following improvements:
+
+* IO reduction: by comparing the `total_scanned_rows` of the TableFullScan operator, you can see that the scan volume of TableFullScan is reduced by 2/3 after Runtime Filter is enabled.
+* Hash join performance improvement: the execution duration of the HashJoin operator is reduced from 376.1 ms to 157.6 ms.
+
+### Best practices
+
+Runtime Filter is applicable to the scenario where a large table and a small table are joined, such as the join query of a fact table and a dimension table. When the dimension table has a small amount of hit data, it means that the filter has fewer values, so the fact table can filter out the data that does not meet the conditions. Compared with the default scenario where the entire fact table is scanned, this significantly improves the query performance.
```
Comment (qiancai) [inline runtime-filter.md:227]:
```suggestion
Runtime Filter is applicable to the scenario where a large table and a small table are joined, such as a join query of a fact table and a dimension table. When the dimension table has a small amount of hit data, it means that the filter has fewer values, so the fact table can filter out the data that does not meet the conditions more effectively. Compared with the default scenario where the entire fact table is scanned, this significantly improves the query performance.
```

File: TOC.md
Comment (qiancai) [inline TOC.md:1]: The translation of changes in system-variables.md is missing

File: runtime-filter.md [hunk 11]
```diff
@@ -0,0 +1,255 @@
...
+
+The Join of the `Sales` table and the `date_dim` table in TPC-DS is a typical example.
+
+## Configure Runtime Filter
+
+When using Runtime Filter, you can configure the mode and predicate type of Runtime Filter.
+
+### Runtime Filter mode
+
+The mode of Runtime Filter is the relationship between the **Filter Sender operator** and **Filter Receiver operator**. There are three modes: `OFF`, `LOCAL`, and `GLOBAL`. In v7.3.0, only `OFF` and `LOCAL` modes are supported. The Runtime Filter mode is controlled by the session-level system variable [`tidb_runtime_filter_mode`](/system-variables.md#tidb_runtime_filter_mode-new-in-v720).
```
Comment (qiancai) [inline runtime-filter.md:237]: tidb_runtime_filter_mode 的 scope 看起来可以是 global 也可以是 session

File: runtime-filter.md [hunk 12]
```diff
@@ -0,0 +1,255 @@
...
+
+- `OFF`: represents that Runtime Filter is disabled. After it is disabled, the query behavior is the same as in previous versions.
+- `LOCAL`: represents that Runtime Filter is enabled in the local mode. In the local mode, the **Filter Sender operator** and **Filter Receiver operator** are in the same MPP Task. In other words, Runtime Filter can be applied to the scenario where the Hash Join operator and Table Scan operator are in the same Task. Currently, Runtime Filter only supports the local mode. To enable this mode, set it to `LOCAL`.
```
Comment (qiancai) [inline runtime-filter.md:240]:
```suggestion
- `LOCAL`: represents that Runtime Filter is enabled in the local mode. In the local mode, the **Filter Sender operator** and **Filter Receiver operator** are in the same MPP task. In other words, Runtime Filter can be applied to the scenario where the HashJoin operator and TableScan operator are in the same task. Currently, Runtime Filter only supports the local mode. To enable this mode, set it to `LOCAL`.
```

File: runtime-filter.md [hunk 13]
```diff
@@ -0,0 +1,255 @@
...
+- `GLOBAL`: currently, the global mode is not supported. You cannot be set Runtime Filter to this mode.
```
Comment (qiancai) [inline runtime-filter.md:241]:
```suggestion
- `GLOBAL`: currently, the global mode is not supported. You cannot set Runtime Filter to this mode.
```

File: runtime-filter.md [hunk 14]
```diff
@@ -0,0 +1,255 @@
...
+
+### Runtime Filter type
+
+The type of Runtime Filter is the type of the predicate used by the generated Filter operator. Currently, only one type is supported: `IN`, which means that the generated predicated is similar to `k1 in (xxx)`. The Runtime Filter type is controlled by the session-level system variable [`tidb_runtime_filter_type`](/system-variables.md#tidb_runtime_filter_type-new-in-v720).
```
Comment (qiancai) [inline runtime-filter.md:245]: tidb_runtime_filter_type 的 scope 看起来可以是 global 也可以是 session

File: runtime-filter.md [hunk 15]
```diff
@@ -0,0 +1,255 @@
...
+
+- `IN`: the default type. It means that the generated Runtime Filter uses the `IN` type predicate.
+
+## Limitations
+
+- Runtime Filter is an optimization in the MPP architecture and can only be applied to queries pushed down to TiFlash.
+- Join type: Left outer, Full outer, and Anti join (when the left table is the Probe Side) do not support Runtime Filter. Because Runtime Filter filters the data involved in the Join in advance, the preceding types of Join do not discard the unmatched data, so this optimization cannot be used.
+- Equal Join expression: When the Probe column in the equal join expression is a complex expression, or when the Probe column type is JSON, Blob, Array, or other complex data types, Runtime Filter is not generated. The main reason is that the preceding types of column are rarely used as the join column. Even if the Filter is generated, the filtering rate is usually low.
```
Comment (qiancai) [inline runtime-filter.md:253]:
```suggestion
- Equal Join expression: When the Probe column in the equal join expression is a complex expression, or when the Probe column type is JSON, Blob, Array, or other complex data types, Runtime Filter is not generated. The main reason is that the preceding types of columns are rarely used as the join column. Even if the Filter is generated, the filtering rate is usually low.
```

File: TOC.md
Comment (ran-huang) [reply TOC.md:1]: 忘了 push，已经推上来了

File: runtime-filter.md [hunk 16]
```diff
@@ -0,0 +1,258 @@
...
+            |         |PhysicalHashJoin   |<-----+
+            |    +----+                   |      |
+4. After RF |    |    +-------------------+      | 1. Scan T2
+    5000    |    |3. Send RF                     |      365
+            |    | filter data                   |
+            |    |                               |
+      +-----+----v------+                +-------+--------+
+      |  TableFullScan  |                | TabelFullScan  |
+      |  store_sales    |                |    date_dim    |
+      +-----------------+                +----------------+
```
Comment (ran-huang) [inline runtime-filter.md:72]:
```suggestion
                         2. Build RF values
            +-------->+-------------------+
            |         |PhysicalHashJoin   |<-----+
            |    +----+                   |      |
4. After RF |    |    +-------------------+      | 1. Scan T2
    5000    |    |3. Send RF                     |      365
            |    | filter data                   |
            |    |                               |
      +-----+----v------+                +-------+--------+
      |  TableFullScan  |                | TabelFullScan  |
      |  store_sales    |                |    date_dim    |
      +-----------------+                +----------------+
```

File: runtime-filter.md [hunk 17]
```diff
@@ -0,0 +1,258 @@
...
+```
+
+*(RF is short for Runtime Filter)*
+
+From the above two figures, you can see that the amount of data scanned by `store_sales` is reduced from 1 million to 5000. By reducing the amount of data scanned by `TableFullScan`, Runtime Filter can reduce the number of times to match the hash table, avoiding unnecessary I/O and network transmission, this significantly improving the efficiency of the join operation.
```
Comment (qiancai) [inline runtime-filter.md:77]:
```suggestion
From the above two figures, you can see that the amount of data scanned by `store_sales` is reduced from 1 million to 5000. By reducing the amount of data scanned by `TableFullScan`, Runtime Filter can reduce the number of times to match the hash table, avoiding unnecessary I/O and network transmission, thus significantly improving the efficiency of the join operation.
```

File: system-variables.md [hunk 18]
```diff
@@ -4306,27 +4306,21 @@ SHOW WARNINGS;
...
->
-> The feature controlled by this variable is not fully functional in the current TiDB version. Do not change the default value.
-
 - Scope: SESSION | GLOBAL
 - Persists to cluster: Yes
 - Type: Enumeration
 - Default value: `OFF`
-- Value options: `OFF`, `LOCAL`
+- Possible values: `OFF`, `LOCAL`
+- Controls the mode of Runtime Filter, that is, the relationship between the **Filter Sender operator** and **Filter Receiver operator**. There are two modes: `OFF` and `LOCAL`. `OFF` means disabling Runtime Filter. `LOCAL` means enabling Runtime Filter in the local mode. For more information, refer to [Runtime Filter mode](/runtime-filter.md#runtime-filter-mode).
```
Comment (qiancai) [inline system-variables.md:4314]:
```suggestion
- Controls the mode of Runtime Filter, that is, the relationship between the **Filter Sender operator** and **Filter Receiver operator**. There are two modes: `OFF` and `LOCAL`. `OFF` means disabling Runtime Filter. `LOCAL` means enabling Runtime Filter in the local mode. For more information, see [Runtime Filter mode](/runtime-filter.md#runtime-filter-mode).
```

File: system-variables.md [hunk 19]
```diff
@@ -4306,27 +4306,21 @@ SHOW WARNINGS;
...
->
-> The feature controlled by this variable is not fully functional in the current TiDB version. Do not change the default value.
-
 - Scope: SESSION | GLOBAL
 - Persists to cluster: Yes
 - Type: Enumeration
 - Default value: `IN`
-- Value options: `IN`
+- Possible values: `IN`
+- Controls the type of predicate used by the generated Filter operator. Currently, only one type is supported: `IN`. For more information, refer to [Runtime Filter type](/runtime-filter.md#runtime-filter-type).
```
Comment (qiancai) [inline system-variables.md:4323]:
```suggestion
- Controls the type of predicate used by the generated Filter operator. Currently, only one type is supported: `IN`. For more information, see [Runtime Filter type](/runtime-filter.md#runtime-filter-type).
```

**Review by elsa0520 — APPROVED** [review]
LGTM

File: runtime-filter.md [hunk 20]
```diff
@@ -0,0 +1,258 @@
...
+      +-----------------+                +----------------+
+```
+
+*(RF is short for Runtime Filter)*
+
+From the above two figures, you can see that the amount of data scanned by `store_sales` is reduced from 1 million to 5000. By reducing the amount of data scanned by `TableFullScan`, Runtime Filter can reduce the number of times to match the hash table, avoiding unnecessary I/O and network transmission, thus significantly improving the efficiency of the join operation.
+
+## Use Runtime Filter
+
+To use Runtime Filter, you need to create a table with TiFlash replicas and set [`tidb_runtime_filter_mode`](/system-variables.md#tidb_runtime_filter_mode-introduced-in-v720) to `LOCAL`.
```
Comment (ran-huang) [inline runtime-filter.md:81]:
```suggestion
To use Runtime Filter, you need to create a table with TiFlash replicas and set [`tidb_runtime_filter_mode`](/system-variables.md#tidb_runtime_filter_mode-new-in-v720) to `LOCAL`.
```

File: runtime-filter.md [hunk 21]
```diff
@@ -0,0 +1,258 @@
...
++--------------+------------+----------+---------------+-----------------+-----------+----------+
+| TABLE_SCHEMA | TABLE_NAME | TABLE_ID | REPLICA_COUNT | LOCATION_LABELS | AVAILABLE | PROGRESS |
++--------------+------------+----------+---------------+-----------------+-----------+----------+
+| tpcds50      | date_dim   |     1015 |             1 |                 |         1 |        1 |
++--------------+------------+----------+---------------+-----------------+-----------+----------+
+```
+
+### Step 2. Enable Runtime Filter
+
+To enable Runtime Filter, set the value of the system variable [`tidb_runtime_filter_mode`](/system-variables.md#tidb_runtime_filter_new-in-v720) to `LOCAL`.
```
Comment (ran-huang) [inline runtime-filter.md:114]:
```suggestion
To enable Runtime Filter, set the value of the system variable [`tidb_runtime_filter_mode`](/system-variables.md#tidb_runtime_filter_mode-new-in-v720  ) to `LOCAL`.
```

**Pull request merged**